src/videojs-http-streaming.js

```diff
@@ -1062,11 +1062,34 @@ class VhsHandler extends Component {
 
     this.player_.tech_.on('keystatuschange', (e) => {
       if (e.status === 'output-restricted') {
-        this.masterPlaylistController_.blacklistCurrentPlaylist({
-          playlist: this.masterPlaylistController_.media(),
-          message: `DRM keystatus changed to ${e.status}. Playlist will fail to play. Check for HDCP content.`,
-          blacklistDuration: Infinity
-        });
+        const masterPlaylist = this.masterPlaylistController_.media();
+        const excludedHDPlaylists = [];
+
+        if (masterPlaylist && masterPlaylist.playlists) {
+          // Assume all HD streams are unplayable and exclude them from ABR selection, then perform
+          // a fastQualityChange to clear the buffer since it may already contain unplayable segments
+          masterPlaylist.playlists.forEach(playlist => {
+            if (playlist && playlist.attributes && playlist.attributes.RESOLUTION &&
+                playlist.attributes.RESOLUTION.height >= 720) {
+              if (!playlist.excludeUntil || playlist.excludeUntil < Infinity) {
+
+                playlist.excludeUntil = Infinity;
+                excludedHDPlaylists.push(playlist);
+              }
+            }
+          });
+
+          if (excludedHDPlaylists.length) {
+            videojs.log.warn(
+              'DRM keystatus changed to "output-restricted." Removing the following HD playlists ' +
+              'that will most likely fail to play and clearing already buffered HD segments. ' +
+              'Check for HDCP content.',
```

Comment:
This may not always be an issue for the stream provider, but can instead involve, for instance, the client not having the right hardware to play back HDCP content. We can add something like:

`This may be due to HDCP restrictions on the stream and the capabilities of the current device."